torch/distributed/tensor/parallel/fsdp.py

```diff
@@ -220,6 +221,66 @@ def _chunk_tensor(
         )
 
 
+def _chunk_dtensor(
+    tensor: torch.Tensor,
+    rank: int,
+    device_mesh: DeviceMesh,
+) -> DTensor:
+    """
+    Shard a tensor to chunks along the first dimension. The local rank will gets its
+    corresponding chunk as the local tensor to create a DTensor.
+    """
+    parent_mesh = mesh_resources.get_parent_mesh(device_mesh)
+    if parent_mesh is None:
+        raise RuntimeError("No parent device_mesh is found for FSDP device_mesh.")
+    if parent_mesh.ndim != 2:
+        raise RuntimeError(
+            f"Found parent device_mesh of ndim={parent_mesh.ndim},",
+            "but only 2D meshes are currently supported.",
+        )
+
+    # We need to explicitly call .detach() to return a new tensor detached from the current graph.
+    tensor = tensor.clone().detach()
+
```

Comment:
not sure why we need the clone/detach as it's not autograd related?

Reply:
> not sure why we need the clone/detach as it's not autograd related?

We need .clone() for sure. From my understanding, the tensor is a view into the FlatParam. If the FlatParam changes, the view will also be changed. The .detach() might not be needed as it should be under @no_grad context from state_dict.  cc. @fegin Correct me if I am wrong.